### PR TITLE
[V2] dns: use getbyname

### LINKED ISSUE #3: Use `inet_res:getbyname` in place of `lookup`
I noticed in the FFI you use `inet_res:lookup`, https://github.com/Pevensie/barnacle/blob/v1.1.0/src/barnacle_ffi.erl#L28

If instead it used `inet_res:getbyname/2`, https://www.erlang.org/doc/apps/kernel/inet_res.html#getbyname/3, it would use the `search` resolver option.

Example, https://github.com/tsloughter/gen_cluster/blob/main/src/gc_discover_dns_a.erl#L36

With `search` it will allow a user to specify in K8s just `<service>.<namespace>` instead of `<service>.<namespace>.svc.cluster.local`.

### PR DESCRIPTION
**Pull request by isaacharrisholt**
Fixes #3

### CODE REVIEW

**Comment by isaacharrisholt** [comment]
@tsloughter I'd appreciate if you could look over this, given you're more Erlang-y than me

File: src/barnacle_ffi.erl [hunk 1]
```diff
@@ -24,8 +26,48 @@ list_local_nodes() ->
...
+  lookup(Name, aaaa, Timeout).
 
-lookup_aaaa(Name) when is_binary(Name) ->
-  inet_res:lookup(binary_to_list(Name), in, aaaa).
+%% Erlang's internal DNS function throw with some inputs like
+%% "ssssrtrssssssssstrtrtrtrdededuddmdidudmdidudmdidudmdidudmddfwddf.com"
+%% so we just catch the error and return an unknown error. I'm not sure
+%% what causes these.
+lookup(Name, Type, Timeout) when is_binary(Name) ->
+  try lookup_throws(Name, Type, Timeout) of
```
Comment (tsloughter) [inline src/barnacle_ffi.erl:40]: Waaa!? Well this is good to know, thanks :)

File: src/barnacle_ffi.erl [hunk 2]
```diff
@@ -24,8 +26,48 @@ list_local_nodes() ->
...
+    Result ->
+      Result
+  catch
+    _:_ ->
+      {error, unknown}
+  end.
+
+lookup_throws(Name, Type, Timeout) when is_binary(Name) ->
+  io:format("Name: ~p~n", [Name]),
```
Comment (tsloughter) [inline src/barnacle_ffi.erl:49]:
```suggestion
```

Think you left this my mistake.

File: src/barnacle/dns.gleam [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    aaaa_records
+    |> list.map(fn(ip) {
+      let #(a, b, c, d, e, f, g, h) = ip
+      IpV6(a, b, c, d, e, f, g, h)
+    })
+
+  Ok(list.concat([a_records, aaaa_records]))
+}
+
+fn format_ip_address(ip: IpAddress) -> String {
```
Comment (tsloughter) [inline src/barnacle/dns.gleam:65]: Do you do this in Gleam instead of calling `inet:ntoa/1` to avoid going back to an Erlang tuple and external ffi?

Comment (isaacharrisholt) [reply src/barnacle/dns.gleam:65]: I mostly didn't know that function existed, but I like to keep things in Gleam where possible.

File: src/barnacle_ffi.erl (hunk 2 above)
Comment (isaacharrisholt) [reply src/barnacle_ffi.erl:49]: Thanks!

**Pull request merged**